base/exports.jl

```diff
@@ -993,6 +993,7 @@ export
     error,
     rethrow,
     retry,
+    default_retry_check,
```

Comment:
users would need to call `default_retry_check` only if they want to change it's keyword arguments.  see the tests for examples.

if your objection is the length of the symbol, there is ample precedent:

```
julia> idx=sortperm(map(x->length(string(x)),names(Base)));
julia> names(Base)[idx[end-20:end]]
21-element Array{Symbol,1}:
 :specialized_binary        
 :task_local_storage        
 :AbstractSparseArray       
 :broadcast!_function       
 :broadcast_setindex!       
 :default_worker_pool       
 :get_zero_subnormals       
 :pointer_from_objref       
 :set_zero_subnormals       
 :AbstractSparseMatrix      
 :AbstractSparseVector      
 :RoundNearestTiesAway      
 :blas_set_num_threads      
 :InvalidStateException     
 :ProcessExitedException    
 :get_bigfloat_precision    
 :set_bigfloat_precision    
 :with_bigfloat_precision   
 :unsafe_pointer_to_objref  
 :specialized_bitwise_unary 
 :specialized_bitwise_binary
```

note that in the parallel realm there is already `default_worker_pool`.  what would you think of `default_check` and `default_pool`, respectively, instead?